src/firefly_exchange_client/client.py

```diff
@@ -48,7 +49,13 @@ async def init(self, user_onboarding=True):
             if 'PERP' in k:
                 self.add_contract(name="Perpetual",address=v["Perpetual"], market=k)
 
-        if user_onboarding:
+        if apiToken:
+            self.apis.api_token = apiToken
+            # for socket
+            self.socket.set_api_token(apiToken)
+            self.webSocketClient.set_api_token(apiToken)
+
+        elif user_onboarding:
```

Comment:
Please add to code comment that when building a read only client, `user_onboarding` is not done.